app/upgrade_cudos.go

```diff
@@ -2032,9 +2032,11 @@ func DoGenesisAccountMovements(genesisData *GenesisData, cudosCfg *CudosMergeCon
 
 		// Handle delegations movement
 		remainingAmountToMove := sdk.NewIntFromBigInt(accountMovement.Amount.BigInt())
+
 		if sourceDelegations, exists := genesisData.Delegations.Get(accountMovement.SourceAddress); exists {
-			for i := range sourceDelegations.Iterate() {
-				validatorAddr, delegatedAmount := i.Key, i.Value
+			// We iterate and delete from source array at the same time
+			for _, validatorAddr := range sourceDelegations.Keys() {
```

Comment:
Are we sure this is the only place (= internals of the whole cycle scope) from which we call delete in the OrderedMap?

Reply:
Yes, but Keys() is now safe to use anyway.

Reply:
`moveGenesisDelegation` is using `Delete` and `moveGenesisDelegation` this is the only place.